app/src/main/contrib/components/Responsibilities/Overview/ConfidentialityAgreement.tsx

```diff
@@ -0,0 +1,110 @@
+
+interface State {
+    checked: boolean;
+}
+
+const fullConfidentialityAgreement = require('./rfp-applicants-confidentiality.pdf');
+
+export class ConfidentialityAgreementComponent extends React.Component<ConfidentialityProps, State> {
+
+    constructor() {
+        super();
+        this.state = {checked: false};
+        this.onChange = this.onChange.bind(this)
+    }
+
+    onChange(e: ChangeEvent<HTMLInputElement>) {
+        console.log(e)
+        this.setState({
+            checked: e.target.checked
+        })
+    }
+
+    componentDidMount() {
+        this.props.getConfidentiality();
+    }
+
+    render() {
+        return <div className={"row"}>
+            <div className={"col-12 col-md-6 offset-md-3"}>
+                <div className={"border p-3 border-dark mb-5"}>
+                    I have read and understood the terms of the <a href={fullConfidentialityAgreement}>
+                    RfP applicants' confidentiality agreement</a>. In doing so, I understand and agree not to disclose
+                    or share any information on vaccine coverage data which I access from Montagu,
+                    beyond my immediate RfP modelling group.
+                    <input type={"checkbox"} className={"mt-2 mb-2 d-block"}
+                           style={{height: "20px", width: "20px"}} checked={this.state.checked}
+                           onChange={this.onChange}/>
+                    {this.state.checked &&
+                    <button className="btn-success" onClick={this.props.signAgreement}>Submit</button>
+                    }
+                </div>
+            </div>
+        </div>
+    }
+
+}
+
+export interface ConfidentialityPublicProps {
+    touchstoneId: string;
+}
+
+interface ConfidentialityPropsFromState extends ConfidentialityPublicProps {
+    signed?: boolean;
+}
+
+interface ConfidentialityProps extends ConfidentialityPropsFromState {
+    signAgreement: () => void;
+    getConfidentiality: () => void;
+}
+
+const mapStateToProps
+    = (state: ContribAppState, props: ConfidentialityPublicProps): ConfidentialityPropsFromState => {
+    return {
+        ...props,
+        signed: state.user.signedConfidentialityAgreement
+    }
+};
+
+const mapDispatchToProps
+    = (dispatch: Dispatch<ContribAppState>, props: ConfidentialityPropsFromState): ConfidentialityProps => {
+    return {
+        ...props,
+        getConfidentiality: () => dispatch(userActionCreators.getConfidentialityAgreement()),
+        signAgreement: () => dispatch(userActionCreators.signConfidentialityAgreement())
+    }
+};
+
+const lifecyleProps: Partial<LifecycleMethods<ConfidentialityProps>> = {
+    onDidMount(props: ConfidentialityProps) {
+        props.getConfidentiality();
+    }
+};
+
+const renderConfidentialityAgreementOrLoadingElement =
+    branch((props: ConfidentialityPropsFromState) => props.signed == false,
+        renderComponent(ConfidentialityAgreementComponent),
```

Comment:
would be nice to have only one branch result here in case if it not stisfy conditions, loading component

Reply:
i noticed branch has one problem, if component has itself triggering of data load, it will not work(such as triggering load on componend load), it only works if you receive props from outside, from parent, or from other component, otherwise it will replace itself with loading component and stop functioning, workaround for this would be using containers pattern: 
https://medium.com/@learnreact/container-components-c0e67432e005

Reply:
Re first comment, unfortunately this can't be simplified in that way because it's a ternary condition: 
1. if the touchstone is non rfp, or the agreement has been signed, render the wrapped component.
Otherwise,
2. if `signed` is null render loading component
3. if `signed` is false, render the agreement form

But maybe there's a way I can make this a bit clearer; maybe using recompose here obfuscates things?

Reply:
I'm afraid I don't understand the second comment. Do you mean that we shouldn't make the call to `getConfidentiality` from this component? The lifecycle method is hooked up before the branching logic, so I don't think this is actually a problem unless I'm missing something? Could you provide steps to replicate the issue you're describing? We could of course move the call to `getConfidentiality` to the pages that use this component. it just means remembering to always make that call before using this HOC.

Reply:
Re the second, I think you're referring to the all to `props.getConfidentiality` in the `ConfidentialityAgreement.componentDidMount` method - that's actually erroneous, so thanks for spotting! The call to `gteConfidentiality` is already mapped to `componentDidMount` in the HOC using recompose (which happens before the branching logic so won't be an issue). I've deleted the obsolete call.

Reply:
Okay, moved the second half of the branching logic out of recompose and into the `ConfidentialityAgreement` component, which hopefully makes this easier to follow!